src/ray/object_manager/pull_manager.cc

```diff
@@ -111,6 +111,9 @@ void PullManager::TryPull(const ObjectID &object_id) {
 
   RAY_LOG(DEBUG) << "Sending pull request from " << self_node_id_ << " to " << node_id
                  << " of object " << object_id;
+  const auto time = get_time_();
+  auto &request = it->second;
+  request.next_pull_time = time + (pull_timeout_ms_ / 1000) * (1 << request.num_retries);
```

Comment:
Doesn't this overflow at some point? We should cap it at some value.

Reply:
Added a cap, but we should never hit it since there's an exponential wait between retries right?